### ISSUE TITLE
Setting ui.select.value when new_value_mode set

### ISSUE THREAD

**Issue by natankeddem** [zauberzeug/nicegui#2033]
If the value isn't in the `ui.select` options the value will not be set. Is this is the intended operation?

Example of setting the value which doesn't work. If you uncomment the commented out lines it will function as intended.
```py
from nicegui import ui

options = ["a", "b", "c"]
select = ui.select(options=options, new_value_mode="add")


def set_select():
    # if input.value not in options:
    #     options.append(input.value)
    select.value = input.value


input = ui.input("value")
ui.button("update", on_click=set_select)

ui.run()
```

**Comment by falkoschindler** [comment]
Thanks for reporting this issue, @natankeddem! We haven't thought about adding a new value programmatically using the `value` setter.

I was just about fixing it by overwriting the `_handle_value_change` method in select.py. But then I realized that the logic for handling new values isn't trivial. If new-value-mode is "add", we basically add a new option. But what if it is "toggle"? Setting `select.value = "foo"` repeatedly will trigger the change handler only once, therefore the select element won't even notice the assignment. This might feel weird and inconsistent.

The implementation for "new-value-mode" is already pretty complex and I'm hesitating to add even more code which would only work in certain situations. Therefore I'm leaning towards only supporting user-induced value changes, which was the original idea. If you need to add an option programmatically, you can always modify the options list directly.

What do you think? Is there a more obvious solution I'm missing? Or should we close it as "won't fix"?

**Comment by natankeddem** [comment]
I think it is fine to leave it as is. Users will have their own preferred logic in the implementation and it might be too much "mind reading" to get a handler that "just works". Maybe this is edge case might be good to add to the docstring/documentation? After, I encountered the erroneous behavior I guessed pretty quickly what was wrong but others might struggle more.
I can try to put more thought into this in the future but right now I don't have much bandwidth. We can always revisit later on if there is a promising solution that appears.

**Comment by drkspace** [comment]
I just ran into this problem when trying to use  `ui.select` inside a reactive method. 

I was able to solve it with

```python
@ui.refreshable
def func():
    val, set_val = ui.state([])
    ui.select([val], new_value_mode='add', value=val, on_change=lambda e: set_val(e.value))
```

It might be worth adding an example in the reactive state docs or `ui.select` docs showing this pitfall. The constructor could also do this automatically, though that wouldn't be intuitive why setting the value in the constructor works but not with `.value` or `.set_value`.
